### PR TITLE
Add getgroups syscall

### PR DESCRIPTION
**Pull request by guzongmin**
The getgroups syscall used by some java applications.
This syscall always return 1 with group ID "0", because Occlum is a single user system.

### CODE REVIEW

File: test/system_info/main.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    printf("machine = %s\n", (const char *)&name.machine);
+    printf("domainname = %s\n", (const char *)&name.__domainname);
+
+    return 0;
+}
+
+static int test_getgroups() {
+    int group_num = getgroups(0, NULL);
+    if (group_num <= 0) {
+        return -1;
```
Comment (tatetian) [inline test/system_info/main.c:23]: Please follow the established practice of using `THROW_ERROR` macro to report errors in test cases.

File: test/system_info/main.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
```
Comment (tatetian) [inline test/system_info/main.c:24]: The check is inconsistent. The later code has made it clear that `group_num` is expected to be `1`. But here the check allows any value of `group_num` greater than `0`.

If we have `group_num == 1`, then the malloc/free of `group_list` just adds needless complexity.

Comment (guzongmin) [reply test/system_info/main.c:24]: Done

File: src/libos/src/process/syscalls.rs [hunk 3]
```diff
@@ -410,3 +410,14 @@ pub fn do_geteuid() -> Result<isize> {
 pub fn do_getegid() -> Result<isize> {
     Ok(0)
 }
+
+// Occlum is a single user enviroment, so only group 0 is supported
+pub fn do_getgroups(size: isize, buf_ptr: *mut u32) -> Result<isize> {
+    if size > 0 && !buf_ptr.is_null() {
```
[the remark this answers is not in the document]
Comment (guzongmin) [reply src/libos/src/process/syscalls.rs]: Done

File: test/system_info/main.c (hunk 1 above)
Comment (guzongmin) [reply test/system_info/main.c]: Done

File: src/libos/src/process/syscalls.rs [hunk 4]
```diff
@@ -410,3 +410,24 @@ pub fn do_geteuid() -> Result<isize> {
...
+// Occlum is a single user enviroment, so only group 0 is supported
+pub fn do_getgroups(size: isize, buf_ptr: *mut u32) -> Result<isize> {
+    if size < 0 {
+        return_errno!(EINVAL, "buffer size is incorrect");
+    } else if size == 0 {
+        //Occlum only has 1 group
+        Ok(1)
+    } else {
+        let size = size as usize;
+        if !buf_ptr.is_null() {
```
[the remark this answers is not in the document]
Comment (guzongmin) [reply src/libos/src/process/syscalls.rs]: Done

**Review by tatetian — APPROVED** [review]
Looks good. Thanks for the contribution!

**Pull request merged**